Bottom: 1f605d0428d31fbb93f2067595178254d3ec1f6d
Top:    3be6bc4eee79f901f71399f2826446a0e06547e6
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-17 19:54:09 +0000

polygon1.c: Refactor poly_Copy0 to use poly_Create(), rather than open-coding

This ensures all polygon initialisation happens in a common place


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 631c3cf..32cfd50 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2827,12 +2827,10 @@ poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
   if (src == NULL)
     return TRUE;
 
-  *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
